components/ratgdo/number/ratgdo_number.cpp

```diff
@@ -10,17 +10,57 @@ namespace ratgdo {
     void RATGDONumber::dump_config()
     {
         LOG_NUMBER("", "RATGDO Number", this);
-        ESP_LOGCONFIG(TAG, "  Type: Rolling Code Counter");
+        if (this->number_type_ == RATGDO_ROLLING_CODE_COUNTER) {
+            ESP_LOGCONFIG(TAG, "  Type: Rolling Code Counter");
+        } else if (this->number_type_ == RATGDO_OPENING_DURATION) {
+            ESP_LOGCONFIG(TAG, "  Type: Opening Duration");
+        } else if (this->number_type_ == RATGDO_CLOSING_DURATION) {
+            ESP_LOGCONFIG(TAG, "  Type: Closing Duration");
+        }
+    }
+
+    void RATGDONumber::set_number_type(NumberType number_type_) 
+    { 
+        this->number_type_ = number_type_; 
+        if (this->number_type_ == RATGDO_OPENING_DURATION || this->number_type_ == RATGDO_CLOSING_DURATION) {
+            this->traits.set_step(0.1);
+        }
     }
 
     void RATGDONumber::on_rolling_code_change(uint32_t rollingCodeCounter)
     {
+        if (this->number_type_ != RATGDO_ROLLING_CODE_COUNTER) {
+            return;
+        }
         this->publish_state(rollingCodeCounter);
     }
+
+    void RATGDONumber::on_opening_duration_change(float duration)
+    {
+        if (this->number_type_ != RATGDO_OPENING_DURATION) {
+            return;
+        }
+        this->publish_state(duration);
+    }
+
+    void RATGDONumber::on_closing_duration_change(float duration)
+    {
+        if (this->number_type_ != RATGDO_CLOSING_DURATION) {
+            return;
+        }
+        this->publish_state(duration);
+    }
+
     void RATGDONumber::control(float value)
     {
-        this->parent_->setRollingCodeCounter(value);
-        this->publish_state(value);
+        if (this->number_type_ == RATGDO_ROLLING_CODE_COUNTER) {
+            this->parent_->setRollingCodeCounter(value);
+        } else if (this->number_type_ == RATGDO_OPENING_DURATION) {
+            this->parent_->setOpeningDuration(value);
+        } else if (this->number_type_ == RATGDO_CLOSING_DURATION) {
+            this->parent_->setClosingDuration(value);
+        }
+        // this->publish_state(value);
```

Comment:
Stale comment?

Reply:
I comment it out because I thought it wasn't necessary and wanted to test. Is it actually needed here and under which conditions?

Reply:
I don't think it is needed. My comment was about a comment being left in

Reply:
Oh, I see. I didn't take it out completely because I wasn't 100% sure it wasn't needed.

Reply:
The client that changes should reflect it but I guess if you have the webserver open and hass it would need to be reflected back as the one that didn't change it probably isn't going to see the change otherwise

Reply:
Yes, I did wonder that too, but I tried with the webserver and hass opened and the change was reflected.